File: src/leetcode/m334/IncreasingTripletSubsequence.java
/*
 * Copyright (c) 2024 THGEEK.COM. All rights reserved.
 */

package leetcode.m334;

/**
 * LeetCode 334
 *
 * @author Tao Hong
 * @version 1.0
 * @since 2024/10/29 23:02
 */
public class IncreasingTripletSubsequence {
    public boolean increasingTriplet(int[] nums) {
        int first = Integer.MAX_VALUE;
        int second = Integer.MAX_VALUE;
        for (int num : nums) {
            if (num <= first) {
                first = num;
            } else if (num <= second) {
                second = num;
            } else {
                return true;
            }
        }
        return false;
    }
}

File: src/leetcode/m334/IncreasingTripletSubsequence.go
package m334

import "math"

// LeetCode 334
func increasingTriplet(nums []int) bool {
	first := math.MaxInt16
	second := math.MaxInt16

	for _, num := range nums {
		if num <= first {
			first = num
		} else if num <= second {
			second = num
		} else {
			return true
		}
	}
	return false
}
